### PR TITLE
Add color picker

### PR DESCRIPTION
**Pull request by h-kanazawa**
#### Which issue does this close?

https://github.com/LearningLocker/enterprise/issues/898

### CODE REVIEW

**Comment by h-kanazawa** [comment]
> 1. I'm not happy with having just the hex code there. Unless you know how to construct them the user will have to go to an external site to create a colour and copy the hex code back into LL. Not great UX.

How about caching all used or recent-used colours in palette.
e.g. We display "default 7 colours" + "org recently using colours (maximum 7 colours)"

> 2. I'm not happy with having just the hex code there. Unless you know how to construct them the user will have to go to an external site to create a colour and copy the hex code back into LL. Not great UX.

How about using picker like "Sketch"?
https://casesandberg.github.io/react-color/
Of course, we can create custom picker by combining react-picker sub components.

> Not obvious that clicking the single circle will open up the menu. If we are going to make this a button, it should look more like a button.

Add a label "Select Colour" (Select a Colour) ??

**Comment by Ian247** [comment]
> > 1. I'm not happy with having just the hex code there. Unless you know how to construct them the user will have to go to an external site to create a colour and copy the hex code back into LL. Not great UX.
> 
> How about caching all used or recent-used colours in palette.
> e.g. We display "default 7 colours" + "org recently using colours (maximum 7 colours)"
> 
I like this idea. Might be worth getting some mock-ups from Ollie as a template to work from.

> > 1. I'm not happy with having just the hex code there. Unless you know how to construct them the user will have to go to an external site to create a colour and copy the hex code back into LL. Not great UX.
> 
> How about using picker like "Sketch"?
> https://casesandberg.github.io/react-color/
> Of course, we can create custom picker by combining react-picker sub components.
> 
I think the Sketch one is more like what I had imagined, but it doesn't quite fit with the LL styling so we would need to customise it a little bit. Will throw this at Ollie for a design. 

> > Not obvious that clicking the single circle will open up the menu. If we are going to make this a button, it should look more like a button.
> 
> Add a label "Select Colour" (Select a Colour) ??
Leave this for now, it might look a bit strange moving the circle to the right in order to have the label and having the label after the circle looks odd.

**Comment by ryansmith94** [comment]
Note to self: Clicking the edit icon created errors in the console and didn't change the UI. Hota is going to investigate. Code looks good.
Screenshot 2019-05-02 at 12 02 30

File: ui/src/components/ColorPicker/CustomColorPicker.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    customColors: React.PropTypes.instanceOf(List),
+
+    // (color: tinycolor.Instance) => void
+    onClickCheckMark: React.PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      color: colorHelper.toState(props.customColors.first('#000000'))
```
Comment (h-kanazawa) [inline ui/src/components/ColorPicker/CustomColorPicker.js]: Default value of immutable.Collection.first() was added on v4.0.0-rc.10. The version `^3.8.1`, we use, does not have it.
https://github.com/immutable-js/immutable-js/releases/tag/v4.0.0-rc.10

**Comment by h-kanazawa** [comment]
@ryanrolds I fixed https://github.com/LearningLocker/learninglocker/pull/1352#issuecomment-488633040

**Comment by HT2-QA** [comment]
This colour picker was used in Curatr and whilst it largely works it several buggy behaviours:
- Selecting the far left hand side (grey/white) will select much further right (bright colour).
- Selecting towards the bottom (black) will select much further up (bright colour).
- Selecting a colour with the slider then dragging the cursor around the shade picker can significantly change the colour selected.

**Comment by h-kanazawa** [comment]
@ryansmith94 I fixed the issues QA team found. Can you review the new commits?

**Comment by ryansmith94** [comment]
Good work and persistence with this one, thanks @h-kanazawa and @HT2-QA 👍

**Pull request merged**